src/arena/simulation.rs

```diff
@@ -198,6 +198,11 @@ impl HoldemSimulation {
         let span = trace_span!("preflop");
         let _enter = span.enter();
 
+        // Notify agents about the start of the betting round
+        for agent in &mut self.agents {
+            agent.handle_round_transition(Round::Preflop);
+        }
```

Comment:
Agents are only supposed to be for making the next action decision. Historians are there to get notifications. That allows for using both together to make the same, but also decouples for others. Meaning agents shouldn't be called here.